src/press/forms.py

```diff
@@ -5,29 +5,46 @@
 
 
 from django import forms
-from django.contrib.contenttypes.models import ContentType
 
 from press import models
-from journal import models as journal_models
-from comms import models as comms_models
+from core.widgets import JanewayFileInput
+from core import files
+from core.middleware import GlobalRequestMiddleware
 
 
 class PressForm(forms.ModelForm):
 
+    press_logo = forms.FileField(
+        required=False,
+        widget=JanewayFileInput,
+    )
+
     class Meta:
         model = models.Press
-        exclude = ('domain', 'thumbnail_image', 'preprints_about', 'preprint_start', 'preprint_pdf_only',
-                   'preprint_submission', 'preprint_publication', 'preprint_editors', 'random_homepage_preprints',
-                   'homepage_preprints')
-        widgets = {'featured_journals': forms.CheckboxSelectMultiple}
-
-    def __init__(self, *args, **kwargs):
-        super(PressForm, self).__init__(*args, **kwargs)
-        press = models.Press.objects.all()[0]
-        content_type = ContentType.objects.get_for_model(press)
-
-        self.fields['featured_journals'].queryset = journal_models.Journal.objects.filter(hide_from_press=False)
-        self.fields['carousel_news_items'].queryset = comms_models.NewsItem.objects.filter(
-            content_type=content_type,
-            object_id=press.pk
+        exclude = (
+            'domain', 'preprints_about', 'preprint_start', 'preprint_pdf_only',
+            'preprint_submission', 'preprint_publication', 'preprint_editors',
+            'random_homepage_preprints', 'homepage_preprints', 'carousel_type',
+            'carousel_items', 'homepage_news_items', 'carousel',
+            'carousel_news_items', 'thumbnail_image',
         )
+        widgets = {
+            'featured_journals': forms.CheckboxSelectMultiple,
+        }
+
+    def save(self, commit=True):
+        press = super(PressForm, self).save(commit=False)
+        request = GlobalRequestMiddleware.get_current_request()
```

Comment:
This feels really clunky, we really need to refactor the files module.

Reply:
It does, and we should!